hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/merkle/MerkleToken.java

```diff
@@ -176,8 +176,18 @@ public boolean equals(final Object o) {
                 && Objects.equals(this.symbol, that.symbol)
                 && Objects.equals(this.name, that.name)
                 && Objects.equals(this.memo, that.memo)
-                && Objects.equals(this.treasury, that.treasury)
-                && Objects.equals(this.autoRenewAccount, that.autoRenewAccount)
+                && Optional.ofNullable(this.treasury).stream()
```

Comment:
This seems massive overkill for a single object comparison.
Alternative:
```java
&& ((this.treasury == null && that.treasury == null)
        || this.treasury.num() == that.treasury.num()))
```
Similar for autoRenewAccount next.